cpp/src/groupby/sort/group_merge_m2.cu

```diff
@@ -44,7 +44,7 @@ namespace {
  */
 template <class result_type>
 struct partial_result {
-  size_type count;
+  result_type count;
```

Comment:
The common approach in libcudf is to use `int64_t` explicitly instead of `size_type` to address size overflow issues. Representing size types with double precision doesn't seem right.

Reply:
Yes but Spark requires it to store in `double`. It will be less efficient to always cast back and forth between `int64_t` vs `double` before and after merging.

Reply:
I see your point, but I'm also concerned about other non-Spark users like cudf-python, cudf-polars, etc. They might end up having to cast types back and forth just to accommodate Spark's needs.

Reply:
Yeah that makes sense. For now I'll change it to `int64_t` for quick fix. We will deal with the casting later.

Reply:
Oh indeed we don't have to compromise. We can have two separate code paths, each processes one type of count so we will all be happy.

Reply:
Done.